### DISCUSSION TITLE
v0.5.0-alpha.2

### DISCUSSION THREAD

**Discussion by buraksezer** [olric-data/olric#147] [Announcements]
Here is the second alpha of v0.5.x tree. It includes many improvements including:

* Integration tests for many packages, especially protocol.
* Distribute client requests in a round-robin fashion.

<hr /><em>This discussion was created from the release <a href='https://github.com/buraksezer/olric/releases/tag/v0.5.0-alpha.2'>v0.5.0-alpha.2</a>.</em>

**Comment by jsierles** [comment]
This version is not working with `go install`, i.e.:

```
 $ go install github.com/buraksezer/olric/cmd/olricd@v0.5.0-alpha.2
go install: github.com/buraksezer/olric/cmd/olricd@v0.5.0-alpha.2 (in github.com/buraksezer/olric@v0.5.0-alpha.2):
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

**Comment by buraksezer** [comment]
Hi @jsierles 

I think that this error is introduced by Go 1.18. There is a minor bug in go-redis/redis package which I don't report yet. I fixed the issue and replaced the dependency with my soft fork.

After a quick investigation, I found this: https://github.com/golang/go/issues/44840. As you can see, there is no proposed solution yet. You can still pull the code and run `go install -v ./cmd/*` to install *olricd*

BTW, Olric 0.5.x is fundamentally different from the previous versions. It uses the Redis protocol and provides a different API. It's still unstable and has many undocumented parts. So sticking with [v0.4.x](https://github.com/buraksezer/olric/tree/v0.4.3) is a good idea during the next months.

**Comment by buraksezer** [comment]
Hi @jsierles, the problem has been fixed in [v0.5.0-alpha.3](https://github.com/buraksezer/olric/discussions/149).
